lib/client.js

```diff
@@ -51,68 +52,36 @@ const DEFAULT_QUERY_COUNT = 100
 const DEFAULT_API_URL = 'https://api.e3db.com'
 const EMAIL = /(.+)@(.+){2,}\.(.+){2,}/
 
-/* eslint-disable no-buffer-constructor */
-
 /**
- * Fallback polyfill to allow for HTTP Basic authentication from either Node
- * or browser-based JavaScript.
+ * Transparent fetch() wrapper to set up authentication headers for either tsv1 or oAuth based authentication
  *
- * @param {string} str String to encode as Base64
- */
-let btoa = function(str) {
-  return new Buffer(str).toString('base64')
-}
-
-/* eslint-enable */
-
-/**
- * Potentially refresh the authorization token used during requests to the E3DB server.
- *
- * The token will be cached for 10 minutes before being automatically refreshed.
- *
- * @param {Config} config E3DB client configuration
+ * @param {Client} client E3DB client instance
+ * @param {string} urlString Absolute URL to fetch from the server
+ * @param {object} options Object representing additional settings for the fetch
  *
- * @returns {Promise<string>}
+ * @returns {Promise}
  */
-async function getToken(client) {
-  if (client._authToken === null || Date.now() > client._authTokenTimeout) {
-    let response
-    try {
-      response = await fetch(client.config.apiUrl + '/v1/auth/token', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'x-www-form-urlencoded',
-          Authorization:
-            'Basic ' + btoa(client.config.apiKeyId + ':' + client.config.apiSecret)
-        },
-        body: 'grant_type=client_credentials'
-      })
-    } catch (err) {
-      console.log('ERROR>>>', err)
-    }
-    let json = await response.json()
-    client._authToken = json.access_token
-    client._authTokenTimeout = Date.parse(json.expires_at)
-
-    return client._authToken
-  }
-  return Promise.resolve(client._authToken)
+async function authenticatedFetch(client, urlString, options) {
```

Comment:
I think these fetch methods belong in the helper object rather than here as methods in the client. The headers generated by the auth header are not generally useful outside of fetch calls, which makes me think the boundary between the helper and client objects are slightly better at the fetch level than the header level. What do you think?

Reply:
Yeah this can live in the helper object, I think that actually makes more sense